crates/code_generation/src/execution.rs

```diff
@@ -247,12 +249,13 @@ fn generate_recording_thread(cyclers: &Cyclers) -> TokenStream {
     quote! {
         {
             let keep_running = keep_running.clone();
+            let parameters_reader = communication_server.get_parameters_reader();
```

Comment:
Is this dropped correctly s.t. nothing leaks and prevents graceful shutdown? Hard to see via GitHub app from my side... 🤪

Reply:
I didn't notice an issue like that, but good point. I'll have a look at this again.

Reply:
The reader is moved into the recording thread and is dropped when the recording thread exits.
We could also fetch the parameters here and then move them by value into the new thread if you prefer.

Reply:
Ok and probably the recording thread terminates when all cyclers terminate. Since this is the same point in time when the other parameter readers are dropped, should be fine then.